agent/consul/options.go

```diff
@@ -24,8 +24,10 @@ type Deps struct {
 type GRPCClientConner interface {
 	ClientConn(datacenter string) (*grpc.ClientConn, error)
 	ClientConnLeader() (*grpc.ClientConn, error)
+	SetGatewayResolver(func(string) string)
```

Comment:
This grpc client pool is created in `agent/setup.go` universally for servers and clients, but only servers will eventually create a GatewayResolver for themselves in `agent/consul/server.go`.  For the regular connection pool for RPCs the gw resolver field is set after the struct was created (but before it's used). I did the same here to avoid having to heavily refactor the agent construction code.